localstack/services/cloudwatch/provider_v2.py

```diff
@@ -183,12 +185,14 @@ def delete_alarms(self, context: RequestContext, alarm_names: AlarmNames) -> Non
         """
         Delete alarms.
         """
-
-        for alarm_name in alarm_names:
-            alarm_arn = arns.cloudwatch_alarm_arn(
-                alarm_name, account_id=context.account_id, region_name=context.region
-            )  # obtain alarm ARN from alarm name
-            self.alarm_scheduler.delete_scheduler_for_alarm(alarm_arn)
+        with _STORE_LOCK:
```

Comment:
q: can I ask what was the issue that necessitated to lock the whole store for the alarms? Is it the size changed during iteration error? or something else?

Reply:
basically the reason what I wrote here: 
> added _STORE_LOCK in the provider, for when we modify the alarm
    reasoning: the set-alarm-state can be triggered any time by the alarm-scheduler, changing the alarm
    this makes concurrent access way more likely
    @baermat I saw in SQS a similar approach, maybe you can share some insights + thoughts on this 🙏

so i didn't encounter any issues during running the tests, but I think it's fair to assume this could become an issue.